src/pages/Blog.jsx

```diff
@@ -0,0 +1,22 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams } from "react-router-dom";
+import { initBlogPostsFromLocalStorage } from "../actions/blogActions";
+import BlogDetails from "../components/BlogDetails/BlogDetails";
+
+const Blog = () => {
+  const { id } = useParams();
+  const blogs = useSelector((state) => state.blog);
+  const blogDetails = blogs.find((blog) => blog.id === id);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(initBlogPostsFromLocalStorage());
+  }, [dispatch]);
+  if (!blogDetails) {
+    return <div>Loading...</div>;
```

Comment:
could you please show a loader, for [example](https://mhnpd.github.io/react-loader-spinner/docs/components/tail-spin) make it a component and show it in the middle of the Blog page

Reply:
implemented from given package